From 2d585743671006180018678f285b9fa944d33cae Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 02 Mar 2023 05:26:28 +0200
Subject: [PATCH] [SW-125667] gaudi2: bist: avoid pattern check in bypass

On some DC setups it was observed that even BIRA passes
for some particular ring, the bypass fails in the pattern
check e.g.

------------x-------------
<13>Feb 14 12:04:03 hls2-bmc83-c06f-kfs ttyOAM3: E: zeros
	check failed, val: 0x480 zeros mask: 0x12490

15:34:03.488
hls2-bmc83-c06f-kfs
<13>Feb 14 12:04:03 hls2-bmc83-c06f-kfs ttyOAM3: E: bira
failed val: 0x25

------------x-------------

DFT team's recommendation is to avoid that pattern check
in bypass. In bypass, the readback value is not guaranteed
to be with expected ones and zeros.
So this patch removes pattern check in bypass step.

Change-Id: I617e2e857d3b32ea2d71a8d3235b79f53bfa7652
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index f856645..543b44c 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -196,12 +196,6 @@
 
 		regmap_writel(GAUDI2_RING_PATTERN_VAL10, mmPSOC_DFT_APB_WORD0);
 		regmap_writel(GAUDI2_RING_PATTERN_VAL7,  mmPSOC_DFT_APB_WORD1);
-
-		expected_ones = 0b00000000000000000000000000000000;
-		expected_zeros = 0b00000000000000000000000000010000;
-		err = gaudi2_sms_read_and_check_pattern(expected_ones, expected_zeros);
-		if (err)
-			ret = 1;
 	}
 
 	return ret;
